controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -311,116 +311,74 @@ bool master_thread::handle_cmdu_1905_1_message(const std::string &src_mac,
 bool master_thread::handle_cmdu_1905_autoconfiguration_search(const std::string &src_mac,
                                                               ieee1905_1::CmduMessageRx &cmdu_rx)
 {
-    std::string al_mac;
-
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_SEARCH_MESSAGE";
 
-    std::shared_ptr<ieee1905_1::tlvAlMacAddressType> tlvAlMacAddressType     = nullptr;
-    std::shared_ptr<ieee1905_1::tlvSearchedRole> tlvSearchedRole             = nullptr;
-    std::shared_ptr<ieee1905_1::tlvAutoconfigFreqBand> tlvAutoconfigFreqBand = nullptr;
-    std::shared_ptr<wfa_map::tlvSupportedService> tlvSupportedServiceIn      = nullptr;
-    std::shared_ptr<wfa_map::tlvSearchedService> tlvSearchedService          = nullptr;
-    int type = cmdu_rx.getNextTlvType();
-
-    while ((type = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
-        switch (type) {
-        case int(wfa_map::eTlvTypeMap::TLV_SUPPORTED_SERVICE):
-            LOG(DEBUG) << "Found TLV_SUPPORTED_SERVICE TLV";
-            tlvSupportedServiceIn = cmdu_rx.addClass<wfa_map::tlvSupportedService>();
-            break;
-        case int(ieee1905_1::eTlvType::TLV_AL_MAC_ADDRESS_TYPE):
-            LOG(DEBUG) << "Found TLV_AL_MAC_ADDRESS_TYPE TLV";
-            tlvAlMacAddressType = cmdu_rx.addClass<ieee1905_1::tlvAlMacAddressType>();
-            break;
-        case int(ieee1905_1::eTlvType::TLV_SEARCHED_ROLE):
-            LOG(DEBUG) << "Found TLV_SEARCHED_ROLE TLV";
-            tlvSearchedRole = cmdu_rx.addClass<ieee1905_1::tlvSearchedRole>();
-            break;
-        case int(ieee1905_1::eTlvType::TLV_AUTOCONFIG_FREQ_BAND):
-            LOG(DEBUG) << "Found TLV_AUTOCONFIG_FREQ_BAND TLV";
-            tlvAutoconfigFreqBand = cmdu_rx.addClass<ieee1905_1::tlvAutoconfigFreqBand>();
-            break;
-        case int(wfa_map::eTlvTypeMap::TLV_SEARCHED_SERVICE):
-            LOG(DEBUG) << "Found TLV_SEARCHED_SERVICE TLV";
-            tlvSearchedService = cmdu_rx.addClass<wfa_map::tlvSearchedService>();
-            break;
-        default:
-            LOG(ERROR) << "Unknown TLV, type " << std::hex << type << ", dropping...";
-            return false;
-            break;
-        }
-    }
-
-    if (tlvAlMacAddressType) {
-        al_mac =
-            network_utils::mac_to_string((const unsigned char *)tlvAlMacAddressType->mac().oct);
-        LOG(DEBUG) << "mac=" << al_mac;
-    } else {
-        LOG(ERROR) << "tlvAlMacAddressType missing - ignoring autconfig search message";
+    auto tlvAlMacAddressType = cmdu_rx.getClass<ieee1905_1::tlvAlMacAddressType>();
+    if (!tlvAlMacAddressType) {
+        LOG(ERROR) << "getClass<tlvAlMacAddressType> failed";
```

Comment:
Really nitpicking here, but one can argue that this is not really an error.
I believe that errors should be reported when something went so wrong that regular operation can't be continued.
In this case, it's probably a malformed CMDU or a missing TLV, which doesn't really impact the Controller's operation, so a WARNING would be more appropriate here.

Reply:
I disagree - I think that errors are anything not expected which changes the way the system behaves. In this case, an agent tried to onboard our controller and we simply didn't answer it due to some missing TLV - we should easilly find this in the logs while looking for either error or warning, so changing to warning is OK, but not as part of this PR since it was ERROR before this change.

IMHO FATAL is more appropriate than error to describe an unrecoverable error, not ERROR.